src/core/amount.js

```diff
@@ -729,7 +705,9 @@ Amount.prototype.parse_native = function(j) {
 // Requires _currency to be set!
 Amount.prototype.parse_value = function(j) {
   this._is_native = false;
-  this._set_value(new BigNumber(j), BigNumber.ROUND_DOWN);
+  const newValue = new IOUValue(
+    new IOUValue(j).toPrecision(16, Value.getBNRoundDown()));
```

Comment:
Yeah, the messiness is the fault of the legacy code that uses two different rounding modes.

The advantage is that we don't have to expose the toPrecision method on the interface. So it's a trade-off between an option on the constructor, vs. an additional method on the interface. I was thinking an option on one method counts as less complexity than a new method, but it's debatable. The other factor is that toPrecision seems less of something that a "Value" should do than having options for how to construct one. What do you think?